tests/checkpoint/test_checkpoint.py

```diff
@@ -1307,3 +1306,145 @@ def test_is_fresh_raises_error_when_child_deps_not_found(in_memory_runtime_conte
     assert diagnostics.success is False
     assert len(diagnostics.errors) == 1
     assert isinstance(diagnostics.errors[0], ValidationDefinitionNotFoundError)
+
+
+@pytest.mark.unit
+def test_is_fresh_raises_errors_for_all_child_validation_definitions(in_memory_runtime_context):
+    context = in_memory_runtime_context
+
+    datasource = context.data_sources.add_pandas(name="my_pandas_ds")
+    asset = datasource.add_dataframe_asset(name="my_pandas_asset")
+    bd_1 = asset.add_batch_definition_whole_dataframe(name="my_bd_1")
+    bd_2 = asset.add_batch_definition_whole_dataframe(name="my_bd_2")
+
+    suite_1 = ExpectationSuite(name="my_suite_1")
+    suite_2 = ExpectationSuite(name="my_suite_2")
+
+    vd_1 = ValidationDefinition(name="my_vd_1", data=bd_1, suite=suite_1)
+    vd_2 = ValidationDefinition(name="my_vd_2", data=bd_2, suite=suite_2)
+
+    cp = Checkpoint(name="my_cp", validation_definitions=[vd_1, vd_2])
+
+    diagnostics = cp.is_fresh()
+    assert diagnostics.success is False
+    assert len(diagnostics.errors) == 5
+    assert [type(err) for err in diagnostics.errors] == [
+        ExpectationSuiteNotAddedError,
+        ValidationDefinitionNotAddedError,
+        ExpectationSuiteNotAddedError,
+        ValidationDefinitionNotAddedError,
+        CheckpointNotAddedError,
+    ]
+
+
+class TestCheckpointPydanticSerializationMethods:
+    """
+    Test overridden Pydantic serialization methods for Checkpoint
+    (dict and json)
+    """
+
+    datasource_name: str = "my_pandas_ds"
+    asset_name: str = "my_pandas_asset"
+    batch_definition_name: str = "my_bd"
+    suite_name: str = "my_suite"
+    validation_definition_name: str = "my_vd"
+    checkpoint_name: str = "my_cp"
+
+    def _build_checkpoint_with_factories(
+        self,
+        context: AbstractDataContext,
+    ) -> Checkpoint:
+        datasource = context.data_sources.add_pandas(name=self.datasource_name)
+        asset = datasource.add_dataframe_asset(name=self.asset_name)
+        bd = asset.add_batch_definition_whole_dataframe(name=self.batch_definition_name)
+        suite = context.suites.add(ExpectationSuite(name=self.suite_name))
+        vd = context.validation_definitions.add(
+            ValidationDefinition(name=self.validation_definition_name, data=bd, suite=suite)
+        )
+        return context.checkpoints.add(
+            Checkpoint(name=self.checkpoint_name, validation_definitions=[vd])
+        )
+
+    def _build_checkpoint_without_factories(self) -> Checkpoint:
+        bd = BatchDefinition(name=self.batch_definition_name)
+        suite = ExpectationSuite(name=self.suite_name)
+        vd = ValidationDefinition(name=self.validation_definition_name, data=bd, suite=suite)
+        return Checkpoint(name=self.checkpoint_name, validation_definitions=[vd])
```

Comment:
If your goal is to test checkpoints with and without factories, you can turn these a parametrized fixture.
https://docs.pytest.org/en/latest/how-to/fixtures.html#fixture-parametrize